.github/workflows/extended.yml

```diff
@@ -87,7 +87,7 @@ jobs:
       - name: Run tests
         run: |
           cd datafusion
-          cargo test  --profile ci --exclude datafusion-examples --exclude datafusion-benchmarks --exclude datafusion-sqllogictest --workspace --lib --tests --features=force_hash_collisions,avro,extended_tests
+          cargo test  --profile ci --exclude datafusion-examples --exclude datafusion-benchmarks --exclude datafusion-sqllogictest --workspace --lib --tests --features=force_hash_collisions,avro,extended_tests --locked
```

Comment:
cargo test shouldn't need the `--locked` flag (here & other places)

it would suffice to verify cargo lock updatodateness once in single job

Reply:
Fail-fast is not unreasonable, i didn't think about this.

otoh one could say same thing about formatting issues. there is definitely tradeoff between conserving CI resources and maxing our single PR feedback signal.
let's put cargo.lock consistency in the "formatting bucket", ie reuse previous design decision for now. i am open to revisiting it.

Reply:
Ok, I'll just keep it for https://github.com/apache/datafusion/blob/684c7df6ecf9c7579654b68cb733a943173961e2/.github/workflows/extended.yml#L37 which is required to pass before a bunch of additional jobs are started.

Reply:
thanks!
can you please add a code comment there why `--locked` is useful?